### PR TITLE
elastic/apm: do not update/use `apm@settings` template

### PR DESCRIPTION
**Pull request by kyungeunni**
Addresses https://github.com/elastic/rally-tracks/pull/550/commits/5c8d4fe310a050adf4d2acd2fbf7cde21e3cbc0f#r1500196363

### CODE REVIEW

File: elastic/apm/index-template.json [hunk 1]
```diff
@@ -4,6 +4,6 @@
     "_meta": {
         "description": "Placeholder template that'd be used when the x-pack-apm car is not used, e.g. in test mode"
     },
-    "composed_of": ["apm@settings"],
+    "composed_of": ["common-apm@custom"],
```
Comment (axw) [inline elastic/apm/index-template.json:7]: What's the reason for defining another non-standard `@custom` component template? Can we use `traces@custom`, `metrics@custom`, and `logs@custom` here?

File: elastic/apm/track.json [hunk 2]
```diff
@@ -10,7 +10,19 @@
   ],
   "component-templates": [
     {
-      "name": "apm@settings",
```
Comment (axw) [inline elastic/apm/track.json:13]: I think you should change these to `traces@custom`, `metrics@custom`, and `logs@custom`. For example, the `metrics-apm.service_destination.1m-*` index template includes `metrics@custom` and `metrics-apm.service_destination.1m@custom`; it does _not_ include `metrics-apm@custom`.

**Review by axw — CHANGES_REQUESTED** [review]

File: elastic/apm/index-template.json (hunk 1 above)
Comment (kyungeunni) [reply elastic/apm/index-template.json]: oops, yes you are right. I should just use the standard custom component.

File: elastic/apm/track.json (hunk 2 above)
Comment (kyungeunni) [reply elastic/apm/track.json:13]: Thanks I updated the PR.

**Review by axw — APPROVED** [review]

File: elastic/apm/index-template.json [hunk 3]
```diff
@@ -1,9 +1,28 @@
...
-    },
-    "composed_of": ["traces@custom", "metrics@custom", "logs@custom"],
-    "priority": 1
+  "index_patterns": ["traces-apm*", "logs-apm*", "metrics-apm*"],
+  "data_stream": {},
+  "_meta": {
+      "description": "Placeholder template that'd be used when the x-pack-apm car is not used, e.g. in test mode"
+  },
+  "composed_of": ["traces@custom", "metrics@custom", "logs@custom"],
+  "priority": 101,
```
Comment (kyungeunni) [inline elastic/apm/index-template.json:8]: setting it between apmplugin template(140) and metrics-*-* template(100)

File: elastic/apm/index-template.json [hunk 4]
```diff
@@ -1,9 +1,28 @@
...
+                "value": {
+                  "type": "long"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
```
Comment (kyungeunni) [inline elastic/apm/index-template.json:27]: @axw I've added this to fix the tests which is there to check the track very quickly for [syntax errors](https://esrally.readthedocs.io/en/stable/adding_tracks.html?highlight=test%20mode#adding-support-for-test-mode). tests were falling due to mapping errors, which shouldn't happen when the proper ingest pipeline/template is installed. In test mode it is not enabled by default so this is a workaround until we enable the apm plugin by default in ES.

**Review by axw — APPROVED** [review]

**Pull request merged**